dart/lib/src/event_processor/url_filter/web_url_filter_event_processor.dart

```diff
@@ -0,0 +1,36 @@
+// We would lose compatibility with old dart versions by adding web to pubspec.
+// ignore: depend_on_referenced_packages
+import 'package:web/web.dart' as web show window, Window;
+
+import '../../../sentry.dart';
+import '../../utils/regex_utils.dart';
+import 'url_filter_event_processor.dart';
+
+UrlFilterEventProcessor urlFilterEventProcessor(SentryOptions options) =>
+    WebUrlFilterEventProcessor(options);
+
+class WebUrlFilterEventProcessor implements UrlFilterEventProcessor {
+  WebUrlFilterEventProcessor(
+    this._options,
+  );
+
+  final web.Window _window = web.window;
+
+  final SentryOptions _options;
+
+  @override
+  SentryEvent? apply(SentryEvent event, Hint hint) {
+    final url = event.request?.url ?? _window.location.toString();
+
+    if (isMatchingRegexPattern(url, _options.allowUrls)) {
+      if (_options.denyUrls.isNotEmpty &&
+          isMatchingRegexPattern(url, _options.denyUrls)) {
+        return null;
+      } else {
+        return event;
+      }
+    } else {
+      return null;
+    }
```

Comment:
can we split this condition up?

something like this

```dart
if (_options.allowUrls.isNotEmpty &&
    !isMatchingRegexPattern(url, _options.allowUrls)) {
  return null;
}

if (_options.denyUrls.isNotEmpty &&
    isMatchingRegexPattern(url, _options.denyUrls)) {
  return null;
}
```

Reply:
I was also unhappy with this code, but your suggestion looks better. I will implement this.